indico/htdocs/js/indico/jquery/categorynavigator.js

```diff
@@ -28,8 +28,19 @@
             openInDialog: false,
             // The title for the category navigator dialog
             dialogTitle: $T.gettext("Select a category"),
-            // Restrict action to categories with no subcategories
-            selectLeafOnly: false,
+            // Allow action to specific categories
+            enableActionOn: {
+                categoriesWithSubcategories: true,
+                categoriesWithEvents: true,
+                self: false
+            },
+            disabledActionMessage: {
```

Comment:
Why not, instead of having two completely separated top-level options, we group the enable state and the disabled message? Something like this:

```js
enableActionOn: {
    categoriesWithSubcategories: {
        disabled: true,
        message: $T.gettext("You cannot select a category that contains subcategories")
    },
    categoriesWithEvents: {
        // ...
    }
    self: {
        // ...
    }
}
```